src/main/scala/com/databricks/sparkdl/python/ModelFactory.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2017 Databricks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.databricks.sparkdl.python
+
+import java.nio.file.{Files, Paths}
+import java.util
+
+import org.apache.log4j.PropertyConfigurator
+
+import scala.collection.JavaConverters._
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.sparkdl_stubs.UDFUtils
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.tensorflow.framework.GraphDef
+import org.tensorframes.{Shape, ShapeDescription}
+import org.tensorframes.impl.{SerializedGraph, SqlOps, TensorFlowOps}
+
+import com.databricks.sparkdl.Logging
+
+/**
+ * Taking over TensorFlow graphs handed from Python
+ * validate the graph 
+ */
+// TODO: merge with the python factory eventually, this is essentially copy/paste
+class GraphModelFactory() extends Logging {
+  private var _shapeHints: ShapeDescription = ShapeDescription.empty
+  // TODO: this object may leak because of Py4J -> do not hold to large objects here.
+  private var _graph: SerializedGraph = null
+  private var _graphPath: Option[String] = None
+  private var _sqlCtx: SQLContext = null
+
+  def initialize_logging(): Unit = initialize_logging("org/tensorframes/log4j.properties")
+
+  /**
+   * Performs some logging initialization before spark has the time to do it.
+   *
+   * Because of the the current implementation of PySpark, Spark thinks it runs as an interactive
+   * console and makes some mistake when setting up log4j.
+   */
+  private def initialize_logging(file: String): Unit = {
+    Option(this.getClass.getClassLoader.getResource(file)) match {
+      case Some(url) =>
+        PropertyConfigurator.configure(url)
+      case None =>
+        System.err.println(s"$this Could not load logging file $file")
+    }
+  }
+
+  /** Setup SQLContext for UDF registeration */
+  def sqlContext(ctx: SQLContext): this.type = {
+    _sqlCtx = ctx
+    this
+  }
+
+  /**
+   * Append shape information to the graph
+   * @param shapeHintsNames names of graph elements
+   * @param shapeHintsShapes the corresponding shape of the named graph elements
+   */
+  def shape(
+    shapeHintsNames: util.ArrayList[String],
```

Comment:
This comment is going to be lost. Mention in the doc of the class that all the public methods are expected to be reachable from python / py4j